src/main/java/me/mrCookieSlime/Slimefun/Objects/Research.java

```diff
@@ -63,11 +63,14 @@
      *            The Cost in XP levels to unlock this {@link Research}
      * 
      */
-    public Research(NamespacedKey key, int id, String name, int defaultCost) {
+    public Research(NamespacedKey key, int id, String name, int defaultCost, SlimefunItem... items) {
         this.key = key;
         this.id = id;
         this.name = name;
         this.cost = defaultCost;
+        
+        if (items.length > 0)
```

Comment:
This length check seems rather unnecessary since an empty array will simply not cause the loop to trigger.